pkg/controller/data_download_controller.go

```diff
@@ -315,10 +315,23 @@ func (r *DataDownloadReconciler) Reconcile(ctx context.Context, req ctrl.Request
 			return ctrl.Result{}, nil
 		}
 
+		// Reserve a slot atomically BEFORE doing expensive GetExposed operation
+		// This prevents race condition where multiple tasks waste time on GetExposed
+		// when the concurrent limit is already reached
+		if err := r.dataPathMgr.ReserveSlot(dd.Name); err != nil {
```

Comment:
Instead of returning error and then you need to check `ConcurrentLimitExceed`, let's just return a boolean for `ReserveSlot`, if it is false, just log the limit reach and requeue.